x/bank/keeper/hooks_test.go

```diff
@@ -0,0 +1,100 @@
+	baseBankKeeper, ok := app.BankKeeper.(keeper.BaseKeeper)
+	require.True(t, ok)
+	keeper.UnsafeSetHooks(
+		&baseBankKeeper, types.NewMultiBankHooks(&bankHooksReceiver),
+	)
+	app.BankKeeper = baseBankKeeper
+
+	// try sending a validSendAmount and it should work
+	err := app.BankKeeper.SendCoins(ctx, addrs[0], addrs[1], validSendAmount)
+	require.NoError(t, err)
+
+	// try sending an invalidSendAmount and it should not work
+	err = app.BankKeeper.SendCoins(ctx, addrs[0], addrs[1], invalidSendAmount)
+	require.Error(t, err)
+
+	// try doing SendManyCoins and make sure if even a single subsend is invalid, the entire function fails
+	err = app.BankKeeper.SendManyCoins(ctx, addrs[0], []sdk.AccAddress{addrs[0], addrs[1]}, []sdk.Coins{invalidSendAmount, validSendAmount})
+	require.Error(t, err)
+
+	// make sure that account to module doesn't bypass hook
+	err = app.BankKeeper.SendCoinsFromAccountToModule(ctx, addrs[0], stakingtypes.BondedPoolName, validSendAmount)
+	require.NoError(t, err)
+	err = app.BankKeeper.SendCoinsFromAccountToModule(ctx, addrs[0], stakingtypes.BondedPoolName, invalidSendAmount)
+	require.Error(t, err)
+
+	// make sure that module to account doesn't bypass hook
+	err = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, stakingtypes.BondedPoolName, addrs[0], validSendAmount)
+	require.NoError(t, err)
+	err = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, stakingtypes.BondedPoolName, addrs[0], invalidSendAmount)
+	require.Error(t, err)
+
+	// make sure that module to module doesn't bypass hook
+	err = app.BankKeeper.SendCoinsFromModuleToModule(ctx, stakingtypes.BondedPoolName, stakingtypes.NotBondedPoolName, validSendAmount)
+	require.NoError(t, err)
+	err = app.BankKeeper.SendCoinsFromModuleToModule(ctx, stakingtypes.BondedPoolName, stakingtypes.NotBondedPoolName, invalidSendAmount)
+	require.Error(t, err)
+
+	// make sure that module to many accounts doesn't bypass hook
+	err = app.BankKeeper.SendCoinsFromModuleToManyAccounts(ctx, stakingtypes.BondedPoolName, []sdk.AccAddress{addrs[0], addrs[1]}, []sdk.Coins{validSendAmount, validSendAmount})
+	require.NoError(t, err)
+	err = app.BankKeeper.SendCoinsFromModuleToManyAccounts(ctx, stakingtypes.BondedPoolName, []sdk.AccAddress{addrs[0], addrs[1]}, []sdk.Coins{validSendAmount, invalidSendAmount})
+	require.Error(t, err)
+
+	// make sure that DelegateCoins doesn't bypass the hook
+	err = app.BankKeeper.DelegateCoins(ctx, addrs[0], app.AccountKeeper.GetModuleAddress(stakingtypes.BondedPoolName), validSendAmount)
+	require.NoError(t, err)
+	err = app.BankKeeper.DelegateCoins(ctx, addrs[0], app.AccountKeeper.GetModuleAddress(stakingtypes.BondedPoolName), invalidSendAmount)
+	require.Error(t, err)
+
+	// make sure that UndelegateCoins doesn't bypass the hook
+	err = app.BankKeeper.UndelegateCoins(ctx, app.AccountKeeper.GetModuleAddress(stakingtypes.BondedPoolName), addrs[0], validSendAmount)
+	require.NoError(t, err)
+	require.Error(t, err)
```

Comment:
This might have been accidentally removed
```suggestion
        err = app.BankKeeper.UndelegateCoins(ctx, app.AccountKeeper.GetModuleAddress(stakingtypes.BondedPoolName), addrs[0], invalidSendAmount)
	require.Error(t, err)
```